### PR TITLE
chore: remove usage of AWSManagedCluster from templates / docs

### PR DESCRIPTION
**Pull request by richardcase**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
This removes the usage of AWSManagedCluster from the templates and docs to help with the deprecation in v1alpha4

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [x] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

**Release note**:

```release-note
Removal of AWSManagedCluster from templates/docs to help with the future deprecation in v1alpha4
```

### CODE REVIEW

File: test/e2e/suites/managed/README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Where possible we re-use an EKS cluster and have tried not to create clusters in
 
 For example, in [eks_test.go](eks_test.go) we perform the following steps:
 
-1. Apply a AWSManagedCluster/AWSManagedControlPlane to create a EKS cluster without any nodes
+1. Apply a AWSManagedControlPlane to create a EKS cluster without any nodes
```
Comment (michaelbeaumont) [inline test/e2e/suites/managed/README.md:9]:
```suggestion
1. Apply an AWSManagedControlPlane to create an EKS cluster without any nodes
```

**Review by michaelbeaumont — COMMENTED** [review]
The tiniest of nits

File: test/e2e/suites/managed/README.md (hunk 1 above)
Comment (richardcase) [reply test/e2e/suites/managed/README.md]: Changed.

**Comment by richardcase** [comment]
The [EKS e2e tests pass](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-aws/2264/pull-cluster-api-provider-aws-e2e-eks/1364915957113294848) with this change

**Pull request merged**